cosmos-tx/src/tx/signer_info.rs

```diff
@@ -0,0 +1,72 @@
+//! Signer info.
+
+use super::{AuthInfo, Fee, ModeInfo, SequenceNumber, SignMode};
+use crate::{crypto::PublicKey, proto, Error, Result};
+use std::convert::{TryFrom, TryInto};
+
+/// [`SignerInfo`] describes the public key and signing mode of a single top-level signer.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct SignerInfo {
+    /// [`PublicKey`] of the signer.
+    ///
+    /// It is optional for accounts that already exist in state. If unset, the verifier can use the
+    /// required signer address for this position and lookup the public key.
+    pub public_key: Option<PublicKey>,
+
+    /// Signing mode.
+    ///
+    /// Nested structure to support nested multisig [`PublicKey`]s.
+    pub mode_info: ModeInfo,
+
+    /// Sequence of the account, which describes the number of committed transactions signed by a
+    /// given address.
+    ///
+    /// It is used to prevent replay attacks.
+    pub sequence: SequenceNumber,
+}
+
+impl SignerInfo {
+    /// Create [`SignerInfo`] for a single direct signer.
+    pub fn single_direct(public_key: Option<PublicKey>, sequence: SequenceNumber) -> SignerInfo {
+        SignerInfo {
+            public_key,
+            mode_info: ModeInfo::single(SignMode::Direct),
+            sequence,
+        }
+    }
+
+    /// Get the [`AuthInfo`] for this signer with the given fee.
+    ///
+    /// This is primarily useful for cases involving a single signer.
+    pub fn auth_info(self, fee: Fee) -> AuthInfo {
+        AuthInfo {
+            signer_infos: vec![self],
+            fee,
+        }
+    }
```

Comment:
Added these as convenience methods for the single direct signer use case, which seems common/important.

The intended usage looks like:

```rust
SignerInfo::single_direct(Some(sender_public_key), sequence_number).auth_info(fee)
```

It's also used in the tests now.